axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceLineServiceImpl.java

```diff
@@ -452,4 +458,82 @@ public Unit getUnit(Product product, boolean isPurchase) {
+    return ObjectUtils.isEmpty(invoiceLineList)
+        ? Boolean.FALSE
+        : invoiceLineList.stream()
+            .anyMatch(
+                invoiceLine ->
+                    invoiceLine.getTypeSelect() == InvoiceLineRepository.TYPE_END_OF_PACK);
+  }
+
+  @Override
+  public boolean isStartOfPackTypeLineQtyChanged(List<InvoiceLine> invoiceLineList) {
+
+    if (ObjectUtils.isEmpty(invoiceLineList)) {
+      return false;
+    }
+    for (InvoiceLine invoiceLine : invoiceLineList) {
+      if (invoiceLine.getTypeSelect() == InvoiceLineRepository.TYPE_START_OF_PACK
+          && invoiceLine.getId() != null) {
+        InvoiceLine oldInvoiceLine = invoiceLineRepo.find(invoiceLine.getId());
+        if (oldInvoiceLine.getTypeSelect() == InvoiceLineRepository.TYPE_START_OF_PACK
+            && invoiceLine.getQty().compareTo(oldInvoiceLine.getQty()) != 0) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  @Override
+  public InvoiceLine updateProductQty(
+      InvoiceLine invoiceLine, Invoice invoice, BigDecimal oldQty, BigDecimal newQty, int scale) {
+    BigDecimal qty =
+        invoiceLine
+            .getQty()
+            .divide(oldQty, scale, RoundingMode.HALF_EVEN)
+            .multiply(newQty)
+            .setScale(scale, RoundingMode.HALF_EVEN);
+    invoiceLine.setQty(qty);
+    if (invoiceLine.getTypeSelect() != InvoiceLineRepository.TYPE_NORMAL
+        || invoiceLine.getProduct() == null) {
+      return invoiceLine;
+    }
+    try {
+      BigDecimal exTaxTotal;
+      BigDecimal inTaxTotal;
+      BigDecimal taxRate = BigDecimal.ZERO;
+      BigDecimal priceDiscounted = this.computeDiscount(invoiceLine, invoice.getInAti());
+      if (invoiceLine.getTaxLine() != null) {
+        taxRate = invoiceLine.getTaxLine().getValue();
+      }
+      if (Boolean.FALSE.equals(invoice.getInAti())) {
+        exTaxTotal = InvoiceLineManagement.computeAmount(qty, priceDiscounted);
+        inTaxTotal = exTaxTotal.add(exTaxTotal.multiply(taxRate));
+      } else {
+        inTaxTotal = InvoiceLineManagement.computeAmount(qty, priceDiscounted);
+        exTaxTotal = inTaxTotal.divide(taxRate.add(BigDecimal.ONE), 2, BigDecimal.ROUND_HALF_UP);
+      }
+      invoiceLine.setExTaxTotal(exTaxTotal);
+      invoiceLine.setCompanyExTaxTotal(this.getCompanyExTaxTotal(exTaxTotal, invoice));
+      invoiceLine.setInTaxTotal(inTaxTotal);
+      invoiceLine.setCompanyInTaxTotal(this.getCompanyExTaxTotal(inTaxTotal, invoice));
+      invoiceLine.setPriceDiscounted(priceDiscounted);
+      invoiceLine.setTaxRate(taxRate);
+    } catch (Exception e) {
+      TraceBackService.trace(e);
+    }
+    return this.computeAnalyticDistributionWithUpdatedQty(invoiceLine);
+  }
+
+  private InvoiceLine computeAnalyticDistributionWithUpdatedQty(InvoiceLine invoiceLine) {
+
+    if (Boolean.FALSE.equals(appAccountService.getAppAccount().getManageAnalyticAccounting())) {
```

Comment:
This condition seems to be incorrect. analyticMoveLineList is not updated when user change qty.